ext/console/02_console.js

```diff
@@ -390,61 +390,85 @@
     const entries = [];
     let iter;
     let valueIsTypedArray = false;
+    let entriesLength;
 
     switch (options.typeName) {
       case "Map":
         iter = MapPrototypeEntries(value);
+        entriesLength = value.size;
         break;
       case "Set":
         iter = SetPrototypeEntries(value);
+        entriesLength = value.size;
         break;
       case "Array":
-        iter = ArrayPrototypeEntries(value);
+        entriesLength = value.length;
         break;
       default:
         if (isTypedArray(value)) {
+          entriesLength = value.length;
           iter = ArrayPrototypeEntries(value);
           valueIsTypedArray = true;
         } else {
           throw new TypeError("unreachable");
         }
     }
 
-    let entriesLength = 0;
-    const next = () => {
-      return iter.next();
-    };
-    while (true) {
-      let el;
-      try {
-        const res = iter.next();
-        if (res.done) {
-          break;
-        }
-        el = res.value;
-      } catch (err) {
-        if (valueIsTypedArray) {
-          // TypedArray.prototype.entries doesn't throw, unless the ArrayBuffer
-          // is detached. We don't want to show the exception in that case, so
-          // we catch it here and pretend the ArrayBuffer has no entries (like
-          // Chrome DevTools does).
-          break;
-        }
-        throw err;
-      }
-      if (entriesLength < inspectOptions.iterableLimit) {
+    if (options.typeName === "Array") {
+      for (
+        let i = 0, j = 0;
+        i < value.length && j < inspectOptions.iterableLimit;
+        i++, j++
```

Comment:
`j` seems unused?
```suggestion
        let i = 0;
        i < MathMin(value.length, inspectOptions.iterableLimit);
        i++
```

Reply:
It's not the same, since `i` is also set with `skipTo`, it will change the behaviour since N consecutive empty values are grouped together and count as 1 entry.

Reply:
I didn't catch that it was mutated within the body of the loop; consider this resolved.